harness/determined/pytorch/_callback.py

```diff
@@ -126,6 +135,16 @@ def on_training_epoch_end(self, epoch_idx: int) -> None:
         """
         pass
 
+    def on_training_workload_end(self, metrics: Dict[str, Any]) -> None:
```

Comment:
is this `metrics` going to contain an extra layer of dictionaries, one for averaged metrics and one for batch metrics?

Reply:
Right now it just the `metrics` arg just receives a single dictionary which has `batch_metrics` and `avg_metrics` as sub-dictionaries, but those could be split out into two args, which is probably clearer. I'll do that.